typescript-loops/main.ts

```diff
@@ -0,0 +1,103 @@
+/* exported getNumbersToTen,
+            getEvenNumbersToTwenty,
+            repeatWord,
+            logEachCharacter,
+            doubleAll,
+            getKeys,
+            getValues
+ */
+function getNumbersToTen(): Array<number> {
+  const numbers = [];
+  let currentNumber = 1;
+  while (currentNumber < 11) {
+    numbers.push(currentNumber);
+    ++currentNumber;
+  }
+  return numbers;
+}
+
+let results1 = [];
+results1 = getNumbersToTen();
+console.log(results1);
+
+function getEvenNumbersToTwenty(): Array<number> {
+  const evenNumbers = [];
+  let currentNumber = 2;
+
+  while (currentNumber < 21) {
+    evenNumbers.push(currentNumber);
+    currentNumber += 2;
+  }
+  return evenNumbers;
+}
+
+const results2 = getEvenNumbersToTwenty();
+console.log('The value of getEvenNumberToTwentyResults:', results2);
+
+function repeatWord(word: string, time: number): string {
+  let count = 1;
+  let repeated = '';
+  while (count < time) {
+    repeated += ' ' + word;
+    ++count;
+  }
+  return repeated;
+}
+
+const results3 = repeatWord('Spider-Man', 10);
+console.log('The value of repeatWordResults:', results3);
+
+function logEachCharacter(str: string): unknown {
+  const a = 0;
+  for (let i = 0; i < str.length; ++i) {
+    console.log(str[i]);
+  }
+  return a;
+}
+logEachCharacter('Spider-Man');
```

Comment:
Remember that not all functions need to `return` a value and not all function `return`s are useful. In this case, since we are just logging the function we don't need a return but the return _type_ of the function would then be `void`

`function logEachCharacter(str: string): void {`